pkgs/development/python-modules/timing-asgi/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, poetry-core
+}:
+
+buildPythonPackage rec {
+  pname = "timing-asgi";
+  version = "0.3.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    pname = "timing_asgi";
+    inherit version;
+    hash = "sha256-XwQLJYhHbASN0J2Mj8S3EVK5RlR09cGJmh93wrQA5G8=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+    --replace poetry.masonry.api poetry.core.masonry.api \
+    --replace "poetry>=" "poetry-core>="
```

Comment:
`--replace-warn` or `--replace-fail` for futureproof-ness

Reply:
Good remark. Fixed.